internal/bodyprocessors/json_test.go

```diff
@@ -116,18 +124,49 @@ var jsonTests = []struct {
 			"json.1.f.0.0":     "1",
 			"json.1.f.0.0.0.z": "abc",
 		},
+		err: nil,
+	},
+	{
+		name: "broken1",
+		json: `{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a": 1 }}}}}}}}}}}}}}}}}}}}}}`,
+		want: map[string]string{},
+		err:  errors.New("invalid JSON"),
+	},
+	{
+		name: "broken2",
+		json: `{"test": 123, "test2": 456, "test3": [22, 44, 55], "test4": 3}`,
+		want: map[string]string{
+			"json.test3.0": "22",
+			"json.test3.1": "44",
+			"json.test3.2": "55",
+			"json.test4":   "3",
+			"json.test":    "123",
+			"json.test2":   "456",
+			"json.test3":   "3",
+		},
+		err: nil,
+	},
+	{
+		name: "bomb",
+		json: strings.Repeat(`{"a":`, deeplyNestedJSONObject) + "1" + strings.Repeat(`}`, deeplyNestedJSONObject),
+		want: map[string]string{
+			"json." + strings.Repeat(`a.`, deeplyNestedJSONObject-1) + "a": "1",
+		},
+		err: errors.New("max recursion reached while reading json object"),
 	},
 }
 
 func TestReadJSON(t *testing.T) {
 	for _, tc := range jsonTests {
 		tt := tc
 		t.Run(tt.name, func(t *testing.T) {
-			jsonMap, err := readJSON(strings.NewReader(tt.json))
-			if err != nil {
-				t.Error(err)
-			}
+			jsonMap, err := readJSON(strings.NewReader(tt.json), maxRecursion)
 			for k, want := range tt.want {
+				if err != nil && err.Error() == tt.err.Error() {
```

Comment:
I think you can extract the first condition and then have a nested if else for `err.Error() == tt.err.Errort()`